legal-api/src/legal_api/services/filings/validations/alteration.py

```diff
@@ -0,0 +1,65 @@
+# Copyright © 2021 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validation for the Alteration filing."""
+from http import HTTPStatus
+from typing import Dict
+
+from flask_babel import _ as babel
+
+from legal_api.errors import Error
+from legal_api.models import Business, Filing
+
+from ... import namex
+from ...utils import get_str
+
+
+def validate(business: Business, filing: Dict) -> Error:
+    """Validate the Alteration filing."""
+    if not business or not filing:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid business and filing are required.')}])
+    msg = []
+
+    nr_path = '/filing/alteration/nameRequest/nrNumber'
+    nr_number = get_str(filing.json, nr_path)
+
+    if (nr_number) {
+        # ensure NR is approved or conditionally approved
+        nr_response = namex.query_nr_number(nr_number)
+        validation_result = namex.validate_nr(nr_response.json())
+        if not validation_result['is_approved']:
+            msg.append({'error': babel('Alteration of Name Request is not approved.'), 'path': nr_path})
```

Comment:
Should the code exit the if block in case of error here (and line 47), or is it OK to keep running to check for more errors?

Reply:
@jachurchill Can you answer this please?

Reply:
Please respond to this question.

Reply:
It certainly could exit here but the requirements didn't say to only make the one check. The other validations can still be done so the error can show that even if the NR was approved it's still not valid for the alteration

Reply:
Sounds good. You said the magic words "the other validations can still be done" :)
(Sometimes other code depends on the previous code and won't work, but as you've thought about this, I'm satisfied 👍 )